File: ignasr-app/src/test/java/limdongjin/ignasr/protos/UserPendingProtoTest.java
package limdongjin.ignasr.protos;

import com.google.protobuf.InvalidProtocolBufferException;
import org.junit.jupiter.api.Test;

import java.util.UUID;

class UserPendingProtoTest {
    @Test
    void test1() throws InvalidProtocolBufferException {
        UserPendingProto.UserPending userPending = UserPendingProto.UserPending
                .newBuilder()
                .setReqId(UUID.randomUUID().toString())
                .setUserId(UUID.randomUUID().toString())
                .build();

        System.out.println(userPending);
        var encoded = userPending.toByteArray();
        UserPendingProto.UserPending decoded = UserPendingProto.UserPending.parseFrom(encoded);
        System.out.println(decoded);
    }
}

File: ignasr-app/src/test/kotlin/limdongjin/ignasr/protos/UserPendingProtoTest.kt
package limdongjin.ignasr.protos

import com.google.protobuf.InvalidProtocolBufferException
import limdongjin.ignasr.protos.UserPendingProto.UserPending
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.util.*

class UserPendingProtoTest {
    @Test
    @Throws(InvalidProtocolBufferException::class)
    fun simpleTest() {
        val reqId = UUID.randomUUID().toString()
        val userId = UUID.randomUUID().toString()
        val userPending = UserPending
            .newBuilder()
            .setReqId(reqId)
            .setUserId(userId)
            .build()

        println(userPending)
        Assertions.assertEquals(reqId, userPending.reqId)
        Assertions.assertEquals(userId, userPending.userId)

        val encoded = userPending.toByteArray()
        val decoded = UserPending.parseFrom(encoded)

        println(decoded)
        Assertions.assertEquals(reqId, decoded.reqId)
        Assertions.assertEquals(userId, decoded.userId)
    }
}